erdjs/src/dapp/walletConnectProvider.ts

```diff
@@ -0,0 +1,198 @@
+import WalletClient from "@walletconnect/client";
+import { IProvider } from "../interface";
+import { Transaction } from "../transaction";
+import { Address } from "../address";
+import { IDappProvider } from "./interface";
+import { Signature } from "../signature";
+
+export class WalletConnectProvider extends EventTarget implements IDappProvider {
+    provider: IProvider;
+    walletConnectBridge: string;
+    address: string = "";
+    walletConnector: WalletClient | undefined;
+    private onWalletConnectLogin: Event = new Event("onWalletConnectLogin");
+    private onWalletConnectDisconect: Event = new Event("onWalletConnectDisconect");
+    constructor(httpProvider: IProvider, walletConnectBridge: string = "") {
+        super();
+        this.provider = httpProvider;
+        this.walletConnectBridge = walletConnectBridge;
+    }
+
+    /**
+     * Initiates wallet connect client.
+     */
+    async init(): Promise<boolean> {
+        this.walletConnector = new WalletClient({
+            bridge: this.walletConnectBridge,
+        });
+        this.walletConnector.on("connect", this.onConnect);
+        this.walletConnector.on("session_update", this.onDisconnect);
+        this.walletConnector.on("disconnect", this.onDisconnect);
+        return true;
+    }
+
+    /**
+     * Returns true if init() was previously called successfully
+     */
+    isInitialized(): boolean {
+        return !!this.walletConnector;
+    }
+
+    /**
+     * Mocked function, returns isInitialized as an async function
+     */
+    isConnected(): Promise<boolean> {
+        return new Promise((resolve, _) => resolve(this.isInitialized()));
+    }
+
+    /**
+     *
+     */
+    async login(): Promise<string> {
+        const chainId = 508;
+        let returnUrl = "";
+        if (!this.walletConnector) {
+            await this.init();
+        }
+        if (this.walletConnector?.connected) {
+            this.walletConnector.killSession();
+        } else {
+            await this.walletConnector?.createSession({ chainId }).then(() => {
+                const uri = this.walletConnector?.uri;
+                if (uri !== undefined) {
+                    returnUrl = uri;
+                }
+            });
+        }
+        return returnUrl;
+    }
+
+    loginAccount = (address: string) => {
+        if (this.addressIsValid(address)) {
+            this.address = address;
+            this.dispatchEvent(this.onWalletConnectLogin);
+        } else {
+            if (this.walletConnector?.connected) {
```

Comment:
Perhaps add some logging, for debugging (optional)?

https://github.com/ElrondNetwork/elrond-sdk/blob/development/erdjs/src/logger.ts

Reply:
Done